### PR TITLE
[AArch64] Upstream PPSSPP's emitter changes.

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Requires a minor change to in the JIT to make sure everything still works.

### CODE REVIEW

File: Source/Core/Common/Arm64Emitter.cpp [hunk 1]
```diff
@@ -1,15 +1,274 @@
...
+	{
+		0x0000000000000001UL,
+		0x0000000100000001UL,
+		0x0001000100010001UL,
+		0x0101010101010101UL,
+		0x1111111111111111UL,
+		0x5555555555555555UL,
+	};
+
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(a[0]))
```
Comment (lioncash) [inline Source/Core/Common/Arm64Emitter.cpp]: Could just make the above table a std::array and get rid of this define.

File: Source/Core/Common/Arm64Emitter.cpp [hunk 2]
```diff
@@ -3032,5 +3673,269 @@ void ARM64FloatEmitter::ABI_PopRegisters(BitSet32 registers, ARM64Reg tmp)
...
+		return false;
+
+	return true;
+}
+
+float FPImm8ToFloat(uint8_t bits)
+{
+	int sign = bits >> 7;
+	uint32_t f = 0;
+	f |= (sign << 31);
```
Comment (lioncash) [inline Source/Core/Common/Arm64Emitter.cpp]: this can just be made a direct assignment.

**Comment by Icekhaos** [comment]
What are we doing with these?

**Comment by Sonicadvance1** [comment]
What?

**Comment by Icekhaos** [comment]
Emitter changes from another emulator.

**Comment by Sonicadvance1** [comment]
The AArch64 emitter that PPSSPP uses is from the Dolphin project.
They've added a few functions and bug fixes to it.
I don't typically try to upstream code from a downstream project(That's their responsibility) but I wanted the codebases to diverge less.

**Comment by Icekhaos** [comment]
Alright, good job. I suggest that, for improving upon Dolphin indirectly, that you keep an eye on PPSSPP for more changes that could benefit us Nintendo folk.

**Pull request merged**